discovery/syncer.go

```diff
@@ -760,7 +774,8 @@ func (g *GossipSyncer) processChanRangeReply(msg *lnwire.ReplyChannelRange) erro
 		// TODO(wilmer): This might require some padding if the remote
 		// node is not aware of the last height we sent them, i.e., is
 		// behind a few blocks from us.
-		if replyLastHeight < queryLastHeight {
+		if replyLastHeight < queryLastHeight &&
+			g.numChanRangeRepliesRcvd < maxReplies {
```

Comment:
So then the main impact of this is that we'll swing back ground to check for if we still need to do a historical sync once we process which round which may _slightly_ slow down IGD? In any case, I would wager the new graph batching logic would make the impact of this change negligible.

Reply:
Correct, if a query results in more replies than expected, then those replies would be delivered later on through our periodical historical syncs.